master/buildbot/secrets/secret.py

```diff
@@ -0,0 +1,69 @@
+# This file is part of Buildbot.  Buildbot is free software: you can
+# redistribute it and/or modify it under the terms of the GNU General Public
+# License as published by the Free Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright Buildbot Team Members
+"""
+secrets as to be used during configuration and run-time
+"""
+class Secret(object):
+    """
+    ...
+    """
+    def __init__(self, vault_name):
+        self._vault_name = vault_name
+
+    @property
+    def vault_name(self):
+        """
+        where the secret is stored
+        """
+        return self._vault_name
+
+
+class ActualSecret(object):
+    """
+    ...
+    """
+    def __init__(self, source, value, props=None):
+        if props is None:
+            props = dict()
+
+        self._source = source
```

Comment:
not really, i think your main comment applies here:

> I think it is best to reason on how to use it rather than how we implement it.
> With this implementation, I have difficulties to get how you would use the SecretInAFile